File: Rust/src/fibonacci.rs
use std::time::Instant;

fn main() {
    let start = Instant::now();
    for i in 1..=50 {
        println!("{} {}", i, fib(i))
    }
    let duration = start.elapsed();
    println!("Finished in: {:?}", duration);
}

fn fib(n: u64) -> u64 {
    match n {
        0 => 0,
        1 => 1,
        _ => fib(n - 1) + fib(n - 2),
    }
}

File: C/fibonacci.c
#include<stdio.h>
#include<sys/time.h>

unsigned long long getTime() {
	struct timeval tv;

	gettimeofday(&tv, NULL);

	unsigned long long millisecondsSinceEpoch =
    (unsigned long long)(tv.tv_sec) * 1000 +
    (unsigned long long)(tv.tv_usec) / 1000;

	return millisecondsSinceEpoch;
}

long long fib(int n) {
	if (n <= 1) {
		return n;
	} else {
		return fib(n-1) + fib(n-2);
	}
}

int main() {
	unsigned long long start = getTime();

    for (int i = 1; i <= 50; i++) {
        printf("%d %lli\n", i, fib(i));
    }

	float time = getTime() - start;
	printf("Completed in: %.3fs", time/1000);

    return 0;
}
